receiver/couchdbreceiver/client.go

```diff
@@ -0,0 +1,125 @@
+package couchdbreceiver
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+)
+
+// CouchDBClient defines the basic HTTP client interface.
+type CouchDBClient interface {
+	BuildReq(path string) (*http.Request, error)
+	Get(url string) ([]byte, error)
+	GetNodeNames() ([]string, error)
+	GetStats(nodeName string) (map[string]interface{}, error)
+}
+
+var _ CouchDBClient = (*couchDBClient)(nil)
+
+type couchDBClient struct {
+	client *http.Client
+	cfg    *Config
+	logger *zap.Logger
+}
+
+// NewCouchDBClient creates a new client to make requests for the CouchDB receiver.
+func NewCouchDBClient(cfg *Config, host component.Host, logger *zap.Logger) (CouchDBClient, error) {
+	client, err := cfg.ToClient(host.GetExtensions())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HTTP Client: %w", err)
+	}
+
+	return &couchDBClient{
+		client: client,
+		cfg:    cfg,
+		logger: logger,
+	}, nil
+}
+
+// Get issues an authorized Get requests to the specified url.
+func (cc *couchDBClient) Get(path string) ([]byte, error) {
+	req, err := cc.BuildReq(path)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := cc.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		if err = resp.Body.Close(); err != nil {
+			cc.logger.Warn("failed to close response body", zap.Error(err))
+		}
+	}()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		if resp.StatusCode >= 400 {
+			cc.logger.Error("couchdb", zap.Error(err), zap.String("status_code", strconv.Itoa(resp.StatusCode)))
+		}
+		return nil, fmt.Errorf("request GET %s failed - %q", req.URL.String(), resp.Status)
+	}
+
+	return body, nil
+}
+
+// Nodes contains a list of all known connected node names.
+type Nodes struct {
```

Comment:
Does `Nodes` need to be exported? Will it be used outside of this package?

Reply:
I am getting an linting error
![Screen Shot 2021-12-16 at 1 05 52 PM](https://user-images.githubusercontent.com/13648435/146425249-7e876876-dd4c-48e5-acc5-89faa525d660.png)

Reply:
this because you named your variable `nodes` and your type is also `nodes` you might need to change your variable name.